Makefile

```
@@ -0,0 +1,49 @@
+# Makefile for Agent Factory
+
+# Configuration
+DOCKER_IMAGE = agent-factory
+DOCKER_CONTAINER = agent-factory-a2a
+DOCKER_TAG = latest
+A2A_SERVER_HOST = localhost
+A2A_SERVER_LOCAL_PORT = 8080
+
+# Default target
+.PHONY: help
+help:
+	@echo "Available targets:"
+	@echo "  build           - Build the Docker image for Agent Factory A2A server"
+	@echo "  run             - Run the A2A server in Docker"
+	@echo "  stop            - Stop the running agent-factory-a2a container"
+	@echo "  clean           - Remove Docker containers and images"
+
+# Build the Docker image
+.PHONY: build
+build:
+	docker build --build-arg APP_VERSION=$(shell git describe --tags --dirty 2>/dev/null || echo "dev") -t $(DOCKER_IMAGE):$(DOCKER_TAG) .
+
+# Run the server in Docker
+.PHONY: run
+run: build
+	@if [ ! -f .env ]; then \
+		echo "Error: .env file not found. Please create one with your API keys."; \
+		exit 1; \
+	fi
+	docker run --rm -d \
+		--name $(DOCKER_CONTAINER) \
+		-p $(A2A_SERVER_LOCAL_PORT):8080 \
+		--env-file .env \
+		$(DOCKER_IMAGE):$(DOCKER_TAG)
+	@echo "Server running at http://$(A2A_SERVER_HOST):$(A2A_SERVER_LOCAL_PORT)"
+
+.PHONY: stop
+stop:
+	@docker stop $(DOCKER_CONTAINER)
```

```suggestion
@if [ "$$(docker ps -q -f name=$(DOCKER_CONTAINER))" ]; then \
        docker stop $(DOCKER_CONTAINER); \
        echo "Successfully stopped $(DOCKER_CONTAINER) container"; \
else \
        echo "No $(DOCKER_CONTAINER) container running"; \
	fi
```

This addresses the following scenario:

1. The user first runs `make stop` to stop the container.
1. The user then runs `make clean` to remove the image.
    1. The `clean` target first runs the `stop` target.
    1. The `stop` target now fails since there's no such container.
    1. The whole `clean` target fails.

---

Makes sense. Updated to use `&&`
```
docker stop $(DOCKER_CONTAINER) && echo "Successfully stopped $(DOCKER_CONTAINER) container";
```
to handle the edge case if `docker stop $(DOCKER_CONTAINER)` fails for some reason it will still `echo "Successfully stopped $(DOCKER_CONTAINER) container";`